app/api/server/v1/commands.js

```diff
@@ -78,15 +82,21 @@ API.v1.addRoute('commands.run', { authRequired: true }, {
 		Meteor.call('canAccessRoom', body.roomId, user._id);
 
 		const params = body.params ? body.params : '';
+		const message = {
+			_id: Random.id(),
+			rid: body.roomId,
+			msg: `/${ cmd } ${ params }`,
+		};
 
-		let result;
-		Meteor.runAsUser(user._id, () => {
-			result = slashCommands.run(cmd, params, {
-				_id: Random.id(),
-				rid: body.roomId,
-				msg: `/${ cmd } ${ params }`,
-			});
-		});
+		if (body.tmid) {
+			const thread = Messages.findOneById(body.tmid);
+			if (!thread || thread.rid !== body.roomId) {
+				return API.v1.failure('Invalid thread.');
```

Comment:
Doesn't this prevent threads from being created? 🤔 Or is there another endpoint for creating threads?